tests/platform/test_reboot.py

```diff
@@ -21,42 +21,75 @@
 from check_interface_status import check_interface_status
 from check_transceiver_status import check_transceiver_basic
 from check_transceiver_status import all_transceivers_detected
+from psu_controller import psu_controller
 
+def check_reboot_cause(dut, reboot_cause_expected):
+    """
+    @summary: Check the reboot cause on DUT.
+    @param dut: The AnsibleHost object of DUT.
+    """
+    logging.info("Check the reboot cause")
+    output = dut.shell("show reboot-cause")
+    reboot_cause_got = output["stdout"]
+    logging.debug("show reboot-cause returns {}".format(reboot_cause_got))
+    m = re.search(reboot_cause_expected, reboot_cause_got)
+    assert m is not None, "got reboot-cause %s after rebooted by %s" % (reboot_cause_got, reboot_cause_expected)
 
-def reboot_and_check(localhost, dut, interfaces, reboot_type="cold"):
+
+def reboot_and_check(localhost, dut, interfaces, reboot_type="cold", reboot_helper=None, reboot_argu=None):
     """
     Perform the specified type of reboot and check platform status.
     """
     logging.info("Run %s reboot on DUT" % reboot_type)
-    if reboot_type == "cold":
-        reboot_cmd = "reboot"
+
+    if reboot_type == "power off":
+        assert reboot_helper is not None, "A reboot function must be provided for hardware reboot"
         reboot_timeout = 300
-    elif reboot_type == "fast":
-        reboot_cmd = "fast-reboot"
-        reboot_timeout = 180
-    elif reboot_type == "warm":
-        reboot_cmd = "warm-reboot"
-        reboot_timeout = 180
+        reboot_cause = "Power Loss"
+        reboot_helper(reboot_argu)
+
+        localhost.wait_for(host=dut.hostname, port=22, state="stopped", delay=10, timeout=120)
     else:
-        assert False, "Reboot type %s is not supported" % reboot_type
-    process, queue = dut.command(reboot_cmd, module_async=True)
-
-    logging.info("Wait for DUT to go down")
-    res = localhost.wait_for(host=dut.hostname, port=22, state="stopped", delay=10, timeout=120,
-        module_ignore_errors=True)
-    if "failed" in res:
-        if process.is_alive():
-            logging.error("Command '%s' is not completed" % reboot_cmd)
-            process.terminate()
-        logging.error("reboot result %s" % str(queue.get()))
-        assert False, "DUT did not go down"
+        if reboot_type == "cold":
```

Comment:
isnt it better to have some kind of a dictionary pre defined with the reboot_type and the relevant parameters and only one line of code to get the info from the dictionary?